### PR TITLE
Fix git cache to do a reset after init.

### PR DESCRIPTION
**Pull request by spxtr**
@krzyzacy this is causing your issue.

### CODE REVIEW

**Comment by spxtr** [comment]
In a future PR I'll use the bootstrap.py from the PR, but until then that's not going to pass.

**Comment by krzyzacy** [comment]
eh, so the previous run does not clean up it's checkout?

**Comment by spxtr** [comment]
That, plus if we delete the working dir and recreate it then it will be empty.

File: jenkins/bootstrap.py [hunk 1]
```diff
@@ -222,6 +222,8 @@ def checkout(call, repo, branch, pull, ssh=False, git_cache=''):
         except OSError:
             pass
         call([git, 'init', repo, '--separate-git-dir=%s' % cache_dir])
+        # Remove any changes left around from previous runs.
+        call([git, 'reset', '--hard'])
```
Comment (fejta) [inline jenkins/bootstrap.py]: Do we also want git add -A to also delete untracked files?

Comment (spxtr) [inline jenkins/bootstrap.py]: git clean, good catch

**Comment by fejta** [comment]
Why not -dfx? 

/lgtm

**Comment by spxtr** [comment]
Done.

**Pull request merged**